packages/client/lib/client/authx/index.ts

```diff
@@ -0,0 +1,13 @@
+import { TokenManager, TokenManagerConfig, TokenStreamListener, RetryPolicy, IDPError } from './token-manager';
+export { TokenManager, TokenManagerConfig, TokenStreamListener, RetryPolicy, IDPError };
```

Comment:
@dhensby Sure, it makes total sense to unify them.

The idea behind the 'barrel' file initially was to create a proper export so users can import from `@redis/client/authx` :

```typescript 
"exports": {
    ".": {
      "types": "./dist/index.d.ts",
      "default": "./dist/index.js"
    }, 
    "./authx": {
      "types": "./dist/lib/client/authx/index.d.ts",
      "default": "./dist/lib/client/authx/index.js"
    },
```
    
But then I found out that we already have imports in peer packages using paths like `@redis/client/lib/` and even `@redis/client/dist/` and many others... This can be addressed in one of these ways:

- Adding more explicit exports (not ideal):

```typescript 
"./index": {
      "types": "./dist/index.d.ts",
      "default": "./dist/index.js"
    },
    "./authx": {
      "types": "./dist/lib/client/authx/index.d.ts",
      "default": "./dist/lib/client/authx/index.js"
    },
    "./lib/commands/generic-transformers": {
      "types": "./dist/lib/commands/generic-transformers.d.ts",
      "default": "./dist/lib/commands/generic-transformers.js"
    },
    "./dist/*": {
      "types": "./dist/*.d.ts",
      "default": "./dist/*.js"
    },
    "./lib/errors": {
      "types": "./dist/lib/errors.d.ts",
      "default": "./dist/lib/errors.js"
    }
```

- Not using exports at all, keeping it as is and creating a barrel file just for user convenience
- Creating a separate submodule containing only the authx support classes

For now, I've decided to stick with not using exports, but I'm considering creating a separate submodule for the authx support classes.